src/main/NWN/API/Object/NwCreature.cs

```diff
@@ -1111,5 +1111,222 @@ public CreatureModelType GetCreatureBodyPart(CreaturePart creaturePart)
      /// </summary>
     public void SetCreatureBodyPart(CreaturePart creaturePart, CreatureModelType creatureModel)
       => NWScript.SetCreatureBodyPart((int)creaturePart, (int)creatureModel, this);
+
+    /// <summary>
+    /// Gets or sets the wing type of this creature.
+    /// </summary>
+    public CreatureWingType WingType
+    {
+      get => (CreatureWingType)NWScript.GetCreatureWingType(this);
+      set => NWScript.SetCreatureWingType((int)value, this);
+    }
+
+    /// <summary>
+    /// Gets or sets the tail type of this creature.
+    /// </summary>
+    public CreatureTailType TailType
+    {
+      get => (CreatureTailType)NWScript.GetCreatureTailType(this);
+      set => NWScript.SetCreatureTailType((int)value, this);
+    }
+
+    /// <summary>
+    /// Get a creature's animal companion creature type.
+    /// </summary>
+    public AnimalCompanionCreatureType AnimalCompanionType(NwCreature creature)
+      => (AnimalCompanionCreatureType)NWScript.GetAnimalCompanionCreatureType(creature);
+
+    /// <summary>
+    /// Gets or sets a value indicating whether this creature will auto-explore the minimap as it walks around.
+    /// </summary>
+    public bool ExploresMinimap
+    {
+      get => NWScript.GetCreatureExploresMinimap(this).ToBool();
+      set => NWScript.SetCreatureExploresMinimap(this, value.ToInt());
+    }
+
+    /// <summary>
+    /// Gets the default level up package.
+    /// </summary>
+    public PackageType StartingPackage
+      => (PackageType)NWScript.GetCreatureStartingPackage(this);
+
+    /// <summary>
+    /// Gets the best talent from a group of talents.
+    /// </summary>
+    public TalentCategory TalentBest(TalentCategory category, int maxCr)
+      => (TalentCategory)NWScript.GetCreatureTalentBest((int)category, maxCr, this);
+
+    /// <summary>
+    /// Retrieves a random talent from a group of talents possessed.
+    /// </summary>
+    public TalentCategory TalentRandom(TalentCategory category)
+      => (TalentCategory)NWScript.GetCreatureTalentRandom((int)category, this);
+
+    /// <summary>
+    /// Gets the PC's familiar type.
+    /// </summary>
+    public FamiliarCreatureType FamiliarType
+      => (FamiliarCreatureType)NWScript.GetFamiliarCreatureType(this);
+
+    /// <summary>
+    /// Gets a value indicating whether this creature was spawned from an encounter.
+    /// </summary>
+    public bool IsEncounter
+      => NWScript.GetIsEncounterCreature(this).ToBool();
+
+    /// <summary>
+    /// Gives xPAmount adjusted by any multiclass penalty.
+    /// </summary>
+    public void GiveXp(int xPAmount)
+      => NWScript.GiveXPToCreature(this, xPAmount);
```

Comment:
This does not work on NPCs, and can be moved into `NwPlayer`.

```suggestion
    /// <summary>
    /// Gives the specified XP to this player, adjusted by any multiclass penalty.
    /// </summary>
    public void GiveXp(int xPAmount)
      => NWScript.GiveXPToCreature(this, xPAmount);
```

Reply:
Moved to `NwPlayer`.  Summary updated.